src/bullet_utils/wrapper.py

```diff
@@ -215,7 +215,8 @@ def get_base_imu_linacc(self):
         base_linvel, base_angvel = pybullet.getBaseVelocity(self.robot_id)
         
         # Transform the base acceleration to the IMU position, in world frame
-        imu_linacc = self.base_linacc + np.cross(self.base_angacc, self.r_base_to_imu) + np.cross(base_angvel, np.cross(base_angvel, self.r_base_to_imu))
+        imu_linacc = self.base_linacc + np.cross(self.base_angacc, rot_base_to_world @ self.r_base_to_imu) +\
+                     np.cross(base_angvel, np.cross(base_angvel, rot_base_to_world @ self.r_base_to_imu))
 
         return self.rot_base_to_imu.dot(rot_base_to_world.T.dot(imu_linacc + np.array([0.0, 0.0, 9.81]))) + self.base_imu_accel_bias + self.base_imu_accel_thermal
```

Comment:
All looks good to me. Does it still work as expected if you change self.rot_base_to_imu to something other than the identity, in the case that the axes of the IMU and base frames are not aligned?

Reply:
Yes, I also ran the ekf in the case which imu is rotated w.r.t base, and the results are correct.